.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,88 @@
+name: Tests
+on:
+  push:
+    tags:
+    - '*'
+    branches:
+    - master
+    - release-*
+  pull_request:
+
+jobs:
+  go:
+    strategy:
+      matrix:
+        gotags: [ 'GOTAGS=""', 'GOTAGS=release' ]
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:stackrox-test-0.3.50
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Cache Go dependencies
+      uses: ./.github/actions/cache-go-dependencies
+
+    - name: Go Unit Tests
+      run: ${{ matrix.gotags }} make go-unit-tests
```

Comment:
I understand what you do is better as you're making tests autonomous and runnable from IDE without having to set `ROX_IMAGE_FLAVOR` environment variable.

However, if this becomes too much of a burden, you can cut that corner and set
```
ROX_IMAGE_FLAVOR=development_build
```
whenever you run `make go-unit-tests`.

The former logic
https://github.com/stackrox/stackrox/blob/6a423ab0693b0cbe4f4208e8d265de3983d832df/scripts/ci/jobs/go-unit-tests.sh#L13-L17
is not relevant because `ROX_IMAGE_FLAVOR=stackrox.io` has no effect any more.